src/Joblify.Core/Offers/OfferService.cs

```diff
@@ -19,13 +23,33 @@ public OfferService(IMapper mapper, IUnitOfWork unitOfWork)
         public async Task<OfferDto> AddOfferAsync(OfferDto offerDto)
         {
             var offer = _mapper.Map<Offer>(offerDto);
-
             await _unitOfWork.OfferRepository.AddAsync(offer);
             await _unitOfWork.CommitAsync();
 
             var created = _mapper.Map<OfferDto>(offer);
 
             return created;
         }
+
+        public async Task<IEnumerable<GetAllCategoriesDto>> GetAllCategories()
+        {
+            var categories = await _unitOfWork.CategoryRepository.Entities.ToListAsync();
+            var getAllCategoriesDtos = _mapper.Map<List<Category>, List<GetAllCategoriesDto>>(categories);
+            return getAllCategoriesDtos;
+        }
+
+        public async Task<IEnumerable<GetAllTradesDto>> GetAllTrades()
+        {
+            var trades = await _unitOfWork.TradeRepository.Entities.ToListAsync();
+            var getAllTradesDtos = _mapper.Map<List<Trade>, List<GetAllTradesDto>>(trades);
+            return getAllTradesDtos;
+        }
+
+        public async Task<GetOfferByIdDto> GetOfferById(int id)
```

Comment:
I'm not sure, but shouldn't we get all offers from index on azure search?

Reply:
Endpoint like this can be used when we want to get full data for certain offer. @Astenna, do we index whole offer object? If yes, then indeed we can just obtain this data straight from search db

Reply:
Yes, we index whole offer object. Let's obtain the data from index